build.cake

```diff
@@ -796,72 +888,91 @@ string GetNuGetPackage(string packageId, string packageVersion)
                     MoveFileToDirectory (nativeFile, targetArchPath);
+                    Console.WriteLine($"Moved native binary file {nativeFile} to {targetArchPath}");
+                } else {
+                    Console.WriteLine("Native binary file already exists");
                 }
             }
         }
-        contentPathSuffix = "runtimes/win10-" + oneArch + "/" + contentPathSuffix;
-    }
-
-    if (package == null) {
-        var files = GetFiles (tempContentPath + contentPathSuffix + "*");
-        MoveFiles (files, destination);
-    }
-}
-
-IList<IPackage> GetNuGetDependencies(IPackageRepository repository, FrameworkName frameworkName, IPackage package)
-{
-    var dependencies = new List<IPackage>();
-    GetNuGetDependencies(dependencies, repository, frameworkName, package);
-    return dependencies;
+        return oneArch;
+    } 
+    return null;
 }
 
-void GetNuGetDependencies(IList<IPackage> dependencies, IPackageRepository repository, FrameworkName frameworkName, IPackage package)
+void MoveAssemblies(NugetDependency package, string tempContentPath, string destination)
 {
-    // Declaring this outside the method causes a parse error on Cake for Mac.
-    string[] IgnoreNuGetDependencies = {
-        "Microsoft.NETCore.UniversalWindowsPlatform",
-        "NETStandard.Library"
-    };
-
-    dependencies.Add(package);
-    foreach (var dependency in package.GetCompatiblePackageDependencies(frameworkName))
+    Console.WriteLine($"MoveAssemblies");
+    var dllFiles = ResolveDllFiles(tempContentPath);
+    if (dllFiles == null)
+    {
+        return;
+    }
+    foreach (var matchingFile in dllFiles) 
     {
-        if (IgnoreNuGetDependencies.Contains(dependency.Id))
+        var targetPath = $"{destination}/{matchingFile.GetFilename()}";
+        if (FileExists(targetPath)) 
         {
-            continue;
+            DeleteFile(targetPath);
         }
-        var subPackage = repository.ResolveDependency(dependency, false, true);
-        if (!dependencies.Contains(subPackage))
-        {
-            GetNuGetDependencies(dependencies, repository, frameworkName, subPackage);
+        MoveFile(matchingFile.FullPath, targetPath);
+        Console.WriteLine($"Moving {matchingFile.FullPath} to {targetPath}");
+    } 
+}
+
+void ExtractNuGetPackage (NugetDependency package, string tempContentPath, string destination)
+{
+    var packageName = package != null ? package.Name : "null";
+    Console.WriteLine($"ExtractNuGetPackage {packageName}; tempcontentpath {tempContentPath}; destination {destination}");
+    var oneArch = MoveNativeBinaries(tempContentPath, destination);
+    if (package != null) {
+        MoveAssemblies(package, tempContentPath, destination);
+    }
+    if (package == null) {
+        var contentPathSuffix = "lib/uap10.0/";
+        if (oneArch != null) {
+            //todo check, the result path may be incorrect
```

Comment:
TODO

Reply:
I checked but didn't find a case where `oneArch` is set to something else with our current set of libraries.
A different set of runtime DLLs are used only in our SQLite lib for now, which is used from a public source only.
cc @annakocheshkova